cmd/server.go

```diff
@@ -40,20 +41,47 @@ var serverCmd = &cobra.Command{
 					},
 				}
 				serverinfos = append(serverinfos, fill...)
-
 			}
 			if idFlag {
 				rowsToDisplay = len(heading)
 			}
 			render.Table(out, heading[:rowsToDisplay], serverinfos)
-
 		} else {
 			render.AsJSON(out, servers)
 		}
 		fmt.Print(out)
 	},
 }
 
+var onCmd, offCmd = &cobra.Command{
+	Use:   "on",
+	Short: "Turn server on using id",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		client.StartServer(ctx, args[0])
+	},
+}, &cobra.Command{
+	Use:   "off",
+	Short: "Turn server off using id",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		if forceFlag {
+			err := client.StopServer(ctx, args[0])
```

Comment:
@c0dehard you can use cmd.Flags().GetString("uuid") instead of args[0]

Reply:
here is an example https://github.com/gridscale/gscloud/blob/develop/cmd/kubernetes.go#L41

Reply:
Like this?
```
id, _ := cmd.Flags().GetString("uuid")
	client.StartServer(ctx, id)
// But it doesn't return the the given uuid, like it originally became from the argument 0
```

and  in init
```
serverCmd.Flags().String("uuid", "", "uuid of a server")
```
By now my code works as it is if i change it like mentioned, in this case i can't power `on` or `off` via an given id because it doesn't get any.
I'm really confused now, it's easier to catch the id via argument in my opinion.
it is not possible for me to implement your suggestion 100% but I understand what you mean in theory.

Reply:
@c0dehard OK. sorry.  Accessing `args` without pre-checking will cause the command to fail `panic: runtime error: index out of range [0] with length 0` if the customer runs in this way`./gscloud server --on`.  What I meant to is keep the command run properly even invalid input and report the right usage message for the customer. If you do the way mentioned https://github.com/gridscale/gscloud/pull/32/files#r439990297, the command format will become `/gscloud server --on --uuid xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`  @bkircher what do you think?

Reply:
Hey Hey :wave:  I think using args is not correct here but I also think flags is not correct here. So let me add a third opinion :laughing: 

From cobra docs: "Commands represent actions, Args are things and Flags are modifiers for those actions."

The thing we want is:
```
$ gscloud server on <id>
$ gscloud server off <id>
$ gscloud server off --force <id>
```

Because on and off are, well, actions and not flags or arguments the should be commands. So there should be

- a `serverOn` command with argument ID
- `serverOff` command with argument ID and flag `--force`

Am I right?

Reply:
In my opinion yes, i still have a problem, i can't catch if theres no flag given
```
argument, err := client.StartServer(ctx, args[0])
		if err != nil {
			panic(err)
		}
```
Does someone see my failure?

Reply:
@c0dehard I would suggest reading https://github.com/spf13/cobra#positional-and-custom-arguments and how to build the custom validator. @bkircher we can keep the first proposed format `gscloud server on <id>`, but we should use the right args. validator.

Reply:
eaa16bc
```
$ ./gscloud server on
```
returns now the help page and:
```
requires at least 1 arg(s), only received 0
```